tree/ntuple/v7/test/ntuple_schemaext.cxx

```diff
@@ -0,0 +1,243 @@
+      std::vector<std::uint32_t> fieldVec;
+      modelUpdater->AddField<std::vector<std::uint32_t>>("vec", &fieldVec);
+      modelUpdater->CommitUpdate();
+
+      modelUpdater->BeginUpdate();
+      float fieldFloat = 10.0;
+      modelUpdater->AddField<float>("f", &fieldFloat);
+      modelUpdater->CommitUpdate();
+
+      modelUpdater->BeginUpdate();
+      std::uint8_t u8 = 0x7f;
+      modelUpdater->AddField<std::uint8_t>("u8", &u8);
+      modelUpdater->CommitUpdate();
+
+      ntuple->Fill();
+      *fieldPt = 12.0;
+      fieldFloat = 1.0;
+      fieldVec = refVec;
+      u8 = 0xaa;
+      ntuple->Fill();
+   }
+
+   auto ntuple = RNTupleReader::Open("myNTuple", fileGuard.GetPath());
+   EXPECT_EQ(2U, ntuple->GetNEntries());
+   EXPECT_EQ(6U, ntuple->GetDescriptor()->GetNFields());
+
+   auto pt = ntuple->GetView<float>("pt");
+   auto vec = ntuple->GetView<std::vector<std::uint32_t>>("vec");
+   auto f = ntuple->GetView<float>("f");
+   auto u8 = ntuple->GetView<std::uint8_t>("u8");
+   EXPECT_EQ(42.0, pt(0));
+   EXPECT_EQ(12.0, pt(1));
+   EXPECT_EQ(std::vector<std::uint32_t>{}, vec(0));
+   EXPECT_EQ(refVec, vec(1));
+   EXPECT_EQ(10.0, f(0));
+   EXPECT_EQ(1.0, f(1));
+   EXPECT_EQ(0x7f, u8(0));
+   EXPECT_EQ(0xaa, u8(1));
+}
+
+TEST(RNTuple, SchemaExtensionProject)
+{
+   FileRaii fileGuard("test_ntuple_schemaext_project.root");
+   std::vector<std::uint32_t> refVec{0x00, 0xff, 0x55, 0xaa};
+   {
+      auto model = RNTupleModel::Create();
+      auto fieldPt = model->MakeField<float>("pt", 42.0);
+
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), "myNTuple", fileGuard.GetPath());
+      auto modelUpdater = ntuple->GetModelUpdater();
+      modelUpdater->BeginUpdate();
+      std::vector<std::uint32_t> fieldVec;
+      modelUpdater->AddField<std::vector<std::uint32_t>>("vec", &fieldVec);
+      modelUpdater->CommitUpdate();
+
+      modelUpdater->BeginUpdate();
+      auto aliasVec = RFieldBase::Create("aliasVec", "std::vector<std::uint32_t>").Unwrap();
+      modelUpdater->AddProjectedField(std::move(aliasVec), [](const std::string &fieldName) {
+         if (fieldName == "aliasVec")
+            return "vec";
+         else
+            return "vec._0";
+      });
+      modelUpdater->CommitUpdate();
+
+      ntuple->Fill();
+      *fieldPt = 12.0;
+      fieldVec = refVec;
+      ntuple->Fill();
+   }
+
+   auto ntuple = RNTupleReader::Open("myNTuple", fileGuard.GetPath());
+   EXPECT_EQ(2U, ntuple->GetNEntries());
+   EXPECT_EQ(6U, ntuple->GetDescriptor()->GetNFields());
+
+   auto pt = ntuple->GetView<float>("pt");
+   auto aliasVec = ntuple->GetView<std::vector<std::uint32_t>>("aliasVec");
+   EXPECT_EQ(42.0, pt(0));
+   EXPECT_EQ(12.0, pt(1));
+   EXPECT_EQ(std::vector<std::uint32_t>{}, aliasVec(0));
+   EXPECT_EQ(refVec, aliasVec(1));
+}
+
+// Based on the RealWorld1 test in `ntuple_extended.cxx`, but here some fields are added after the fact
+TEST(RNTuple, SchemaExtensionRealWorld1)
```

Comment:
Ok for now. In the final version, let's find a real-world test that mimics the use in experiment frameworks.